backend/libinput/keyboard.c

```diff
@@ -23,9 +23,17 @@ static void wlr_libinput_keyboard_destroy(struct wlr_keyboard *wlr_kb) {
 	libinput_device_unref(wlr_libinput_kb->libinput_dev);
 }
 
+void wlr_libinput_keyboard_update_state(struct wlr_keyboard *keyboard,
+		struct wlr_event_keyboard_key *event) {
+	uint32_t keycode = event->keycode + 8;
+	xkb_state_update_key(keyboard->xkb_state, keycode,
+		event->state == WLR_KEY_PRESSED ? XKB_KEY_DOWN : XKB_KEY_UP);
```

Comment:
I don't like how xkb stuff is mixed with libinput stuff.

Reply:
I agree. This shouldn't be here.